exporter/otlpexporter/mock_receiver.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otlpexporter // import "go.opentelemetry.io/collector/exporter/otlpexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
+)
+
+// DataReceiverBase implement basic functions needed by all receivers.
+type DataReceiverBase struct {
```

Comment:
Why is this needed as a separate struct?

Reply:
I used the structure from contrib's testbed, but it indeed does not help here, I deleted this part and port is set up by the default config of otlp receiver factory